mitosheet/src/mito/utils/actions.tsx

```diff
@@ -714,6 +784,58 @@ export const getActions = (
             searchTerms: ['filter', 'remove', 'delete'],
             tooltip: "Filter this dataframe based on the data in a column."
         },
+
+        [ActionEnum.FilterToCellValue]: {
+            type: 'build-time',
+            staticType: ActionEnum.FilterToCellValue,
+            titleContextMenu: 'Filter to Cell Value',
+            iconContextMenu: FilterIcon,
+            longTitle: 'Filter column',
+            actionFunction: async () => {
+                // We turn off editing mode, if it is on
+                setEditorState(undefined);
+                const {startingColumnIndex} = gridState.selections[0];
+                const columnID = sheetData.data[startingColumnIndex].columnID;
+                const { columnFilters, cellValue, columnDtype } = getCellDataFromCellIndexes(sheetData, gridState.selections[0].startingRowIndex, gridState.selections[0].startingColumnIndex);
+                const filters = columnFilters?.filters ?? [];
+                
+                let condition = 'string_exactly';
+                if (columnDtype === 'number') {
+                    condition = 'number_exactly';
+                } else if (columnDtype === 'boolean') {
+                    condition = cellValue ? 'boolean_is_true' : 'boolean_is_false';
+                } else if (columnDtype === 'datetime') {
+                    condition = 'datetime_exactly';
+                }
+                filters.push({
+                    condition: condition,
+                    value: cellValue
+                } as FilterType);
```

Comment:
Check out the function `getExclusiveFilterData`. It implements this logic, while handling NaN values as well. 

I forgot about it until I saw this logic...